c'est une proposition de correction/refactoring

VincentProuchet opened the pull request:
UserEntity
User-get.dto
  modification des propriètées pour permettre une valeur null
  à celle acceptées par la base de données

user.service
  lors du changement de mot de passe dans une procédure forgotpwd
  le token resetPwdToken est mit à null au lieu de undefined
  car undefined n'est pas reconnu par les bases de données
  et typeORM ne semble pas faire la conversion

VincentProuchet committed:
Correction/refactoring

UserEntity
User-get.dto
  modification des propriètées pour permettre une valeur null
  à celle acceptées par la base de données

user.service
  lors du changement de mot de passe dans une procédure forgotpwd
  le token resetPwdToken est mit à null au lieu de undefined
  car undefined n'est pas reconnu par les bases de données
  et typeORM ne semble pas faire la conversion

VincentProuchet committed:
UserGetDto

  remis en place la possibilitées de certaines propriètées
   à être undefined

Guuzii commented on src/user/dto/out/user-get.dto.ts:
```diff
@@ -1,17 +1,17 @@
-  firstname?: string;
-  lastname?: string;
-  avatarUrl?: string;
+  firstname: string | null;
```
if faudrait remettre les "?" dans l'interface pour rendre les propriété optionnelles

Guuzii commented on src/user/dto/out/user-get.dto.ts:
```diff
@@ -1,17 +1,17 @@
-  firstname?: string;
-  lastname?: string;
-  avatarUrl?: string;
+  email?: string;
```
la propriété email de doit pas être optionnelle, retirer le "?"

VincentProuchet replied:
modification demandée faite et push

Guuzii commented on src/user/entities/user.entity.ts:
Même chose que pour le dto, il faut laisser les propriétés optionelles (remettre les "?")

VincentProuchet replied:
modifications demandées faites et push

VincentProuchet committed:
remis ?

remis ? sur les propriètées optionnelles de
 userGetDto
 IUserGetDto
retrait du ? sur la propriète email de userGetDto
